apps/server/src/modules/user/repo/role.repo.ts

```diff
@@ -0,0 +1,110 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { Role, IPermissionsAndRoles } from '../entity';
+
+interface CacheEntry {
+	[name: string]: { value: Role; validUntil: number };
+}
+
+// TODO: no query handled at the moment
+// TODO: write test and move
+// TODO: replace with generall solution if it is added
+export class RoleCache {
```

Comment:
the microorm has a simple cache mechanism built in, see https://mikro-orm.io/docs/caching

Reply:
I am happy if we can replace it, is only copy paste from legacy server.